runtime/parachains/src/ump.rs

```diff
@@ -52,6 +52,37 @@ impl UmpSink for () {
 	}
 }
 
+/// A specific implementation of a UmpSink where messages are in the XCM format
+/// and will be forwarded to the XCM Executor.
+pub struct XcmSink<Config>(sp_std::marker::PhantomData<Config>);
+
+impl<Config: xcm_executor::Config> UmpSink for XcmSink<Config> {
+	fn process_upward_message(origin: ParaId, msg: Vec<u8>) -> Weight {
+		use parity_scale_codec::Decode;
+		use xcm::VersionedXcm;
+		use xcm::v0::{Junction, MultiLocation, ExecuteXcm};
+		use xcm_executor::XcmExecutor;
+
+		let weight: Weight = 0;
+
+		if let Some(versioned_xcm_message) = VersionedXcm::decode(&mut &msg[..]).ok() {
```

Comment:
```suggestion
		if let Ok(versioned_xcm_message) = VersionedXcm::decode(&mut &msg[..]) {
```

And maybe some logging when it fails? :see_no_evil: